source/client_channel_handler.c

```diff
@@ -199,10 +199,15 @@ static int s_packet_handler_connack(
             (void *)connection);
         MQTT_CLIENT_CALL_CALLBACK_ARGS(
             connection, on_connection_complete, AWS_OP_SUCCESS, connack.connect_return_code, connack.session_present);
-        MQTT_CLIENT_CALL_CALLBACK_ARGS(
-            connection, on_connection_success, connack.connect_return_code, connack.session_present);
     }
 
+    /*
+     * The on_connection_success would get triggered on the successful CONNACK. It invoked with both the first connect
+     * attempt and reconnection attempt as Mqtt5 does not have on_resume callback for reconnection.
+     */
+    MQTT_CLIENT_CALL_CALLBACK_ARGS(
```

Comment:
We might need an on_connection_failure callback above in the line 170 code block that returns.

Reply:
nvm, invoked during the shutdown process =P